### DISCUSSION TITLE
How to prerender after fetching data

### DISCUSSION THREAD

**Discussion by SleepiestAdam** [nuxt/nuxt#16238] [Questions]
So I'm trying to prerender dynamic routes of my site and broadly setup the entire site to run using swr.

I've got it adding the dynamic routes using the nitro:config hook and it is prerendering all my routes.

However; my app, on launch hits up our REST api to get all the content for the site (it's a site that provides bedtime stories, so basically an array of our content, cover images, descriptions, that kind of stuff)... It does it all in one fetch on launch, async lazily.

I have accounted for loading time on all pages (to account for any new stories that have been added that have not yet been pre-rendered), however my aim with the prerendered routes is to have it render the majority of routes "after the data has been collected" as for the vast majority of the content pages this page is unlikely to update very often once the given bedtime story has been added to the system.

Unfortunately it's pre-rendering the pages before it's got the async data back from our API, which does provide a little benefit in that the loading screen loads faster, but isn't what I really want it to do.

My question is - how can I go about making it wait until the data fetch request has been completed before pre-rendering the routes / is there any way to segment out the behaviour for pre-rendering so that when pre-rendered it will fetch the data before pre-rendering each route, but have it still load the data lazily in the background when deployed in production?

**Comment by TheAlexLichter** [comment]
Hard to tell without code. As long as you await `useFetch`/ `useAsyncData`, data should be available on the server-side and part of the HTML sent to the client.
